chat.js

```diff
@@ -2,34 +2,23 @@ import React from "react";
 import ReactDOM from "react-dom";
 import route from "can-route";
 import DefineMap from "can-define/map/";
-import Component from "react-view-model/component";
 import Messages from "./components/messages";
+import { connect, ObserveObject } from "ylem";
 
-export const ViewModel = DefineMap.extend('AppVM', {
-	page: "string",
-	message: {
-		type: "string",
-		value: "Chat Home",
-		serialize: false
-	}
-});
+class Store extends ObserveObject { 
 
-export default class App extends Component {
-	addExcitement() {
-		this.viewModel.message = this.viewModel.message + "!";
-	}
+}
 
-	render() {
-		return (
+const Chat = () => {
+	return(
 			<div className="container">
 				<div className="row">
 					<div className="col-sm-8 col-sm-offset-2">
-						{ this.viewModel.page === "home" ? (
+						{ route.data.page === "home" ? (
```

Comment:
Since route is an external store, I would not consume it directly in the component. Either pass it in as a prop, or consume it directly from the VM and pass that value to the component with a getter.

One of the design goals is that the component (even if you never do) can be used entirely outside ylem; it is *just* a plain component.

Edit: now that I see the connection here, you should be able to use `this.props.page` (after you hook up the routing to use the instance).